tiledb/sm/storage_manager/consolidator.cc

```diff
@@ -64,12 +64,21 @@ Status Consolidator::consolidate(const char* array_name) {
   std::vector<URI> old_fragment_uris;
   URI array_uri = URI(array_name);
 
-  // Open array
+  // Open array for reading
   auto open_array_for_reads = (OpenArray*)nullptr;
-  auto open_array_for_writes = (OpenArray*)nullptr;
   uint64_t snapshot;
   RETURN_NOT_OK(storage_manager_->array_open(
       array_uri, QueryType::READ, &open_array_for_reads, &snapshot));
+  open_array_for_reads->mtx_lock();
```

Comment:
Perhaps it is better to move the mutex lock in `OpenArray::is_empty`?

Reply:
I suggest we move it to `OpenArray::is_open` and later or we do the same for the other methods as well.